frontend/src/pages/featureStore/screens/featureServices/FeatureServicesListView.tsx

```diff
@@ -34,28 +36,73 @@ const FeatureServicesListView = ({
     [setFilterData],
   );
 
-  const onClearFilters = React.useCallback(() => setFilterData({}), [setFilterData]);
+  const onClearFilters = React.useCallback(() => {
+    setFilterData({});
+    setTagFilters([]);
+  }, []);
 
-  const filteredFeatureServices = React.useMemo(
-    () =>
-      applyFeatureServiceFilters(
-        processedFeatureServices,
-        featureServices.relationships,
-        filterData,
-      ),
-    [processedFeatureServices, featureServices.relationships, filterData],
-  );
+  const handleTagClick = React.useCallback((tagString: string) => {
+    setTagFilters((prevTags) => {
+      if (!prevTags.includes(tagString)) {
+        return [...prevTags, tagString];
+      }
+      return prevTags;
+    });
+    setCurrentFilterType('tag');
+  }, []);
+
+  const handleTagFilterRemove = React.useCallback((tagToRemove: string) => {
+    setTagFilters((prevTags) => prevTags.filter((tag) => tag !== tagToRemove));
+  }, []);
+
+  const handleTagFilterAdd = React.useCallback((newTag: string) => {
+    setTagFilters((prevTags) => {
+      if (!prevTags.includes(newTag)) {
+        return [...prevTags, newTag];
+      }
+      return prevTags;
+    });
+  }, []);
+
+  const handleFilterTypeChange = React.useCallback((filterType: string) => {
+    setCurrentFilterType(filterType);
+  }, []);
```

Comment:
These 4 functions are declared for every file. better we export and make these as utility.